sdk/communication/azure-communication-callingserver/azure/communication/callingserver/_server_call.py

```diff
@@ -45,14 +52,129 @@ def play_audio(
         if not play_audio_options:
             raise ValueError("options can not be None")
 
-        play_audio_request = PlayAudioRequestConverter.convert(audio_file_uri, play_audio_options)
+        if not CallingServerUtils.is_valid_url(audio_file_uri):
+            raise ValueError("audio_file_uri is invalid")
+
+        if not play_audio_options.audio_file_id:
+            raise ValueError("audio_file_id can not be None")
+
+        if not CallingServerUtils.is_valid_url(play_audio_options.callback_uri):
+            raise ValueError("callback_uri is invalid")
+
+        if not play_audio_options.operation_context:
+            raise ValueError("operation_context can not be None")
+
+        play_audio_request = PlayAudioRequestConverter.convert(
+            audio_file_uri, play_audio_options)
 
         return self._server_call_client.play_audio(
             server_call_id=self.server_call_id,
             request=play_audio_request,
             **kwargs
         )
 
+    @distributed_trace()
+    def start_recording(
+        self,
+        server_call_id,  # type: str
```

Comment:
server_call_id is not needed in any of these calls given that it is already part of the object itself, plus you are not using them because of the same reason.